Update security page and security guide

lukasbestle opened the pull request:
In early preparation for the update check feature.

lukasbestle committed:
Security guide: Explain EOL/vulnerable versions

Link destination for the EOL warning in the update check

texnixe commented on content/docs/1_guide/19_security/guide.txt:
```diff
@@ -31,6 +31,16 @@ Kirby's current version number is stored in `/kirby/composer.json` in the versio
+
+Whenever we find or get notified of a vulnerability, we release updates for all Kirby versions with active support and security support. Active support means that the version receives functionality updates, enhancements and bug fixes. Security support means that the version only receives security updates.
+
+Versions that are "end of life" receive no updates whatsoever. We cannot ensure that these versions are free of vulnerabilities and security researchers generally don't review these versions for vulnerabilities either.
```
What I'm wondering Shouldn't 3.* versions before a security fix then have an EOL tag as well, instead of 3.* => active-support, at least that might be a bit misleading.

lukasbestle replied:
I have written this guide section so that it can be linked to from the security issue list in the Panel System view. So users who get sent here are on an older major release that will not receive any more updates. Whereas users who are on 3.x but on a release with a known vulnerability won't get the EOL warning but a specific warning about the vulnerability instead, also with the recommendation to update.

(Of course the update check will only be implemented in 3.x, so it won't be displayed on v1 or v2. But if we ever release v4 and v3 is EOL, the EOL check will become active. All these thoughts are planning ahead.)

The data in the table is structured for the update check logic. I have updated the guide wording though to be more clear. Is it better now?

texnixe replied:
Yes, it's fine!

lukasbestle committed:
Security Incidents: Update old descriptions

More consistent with the description format of newer incidents (weakness type at the beginning)

lukasbestle committed:
Security: Rename `Supported` field to `Versions`

Better matches the purpose of the information that goes beyond just the support status

lukasbestle committed:
Security: Add docs for the security data structure

lukasbestle committed:
Security Incidents: Single link per incident

More machine-readable data structure (we never used additional links so far anyway)

lukasbestle committed:
Security Versions: Move to key-value structure

Allows direct access to e.g. the latest version

lukasbestle committed:
Security Versions: Include latest patch for majors

Useful for update checks (allows to find the latest free update release for each major release)

lukasbestle committed:
Security Incidents: Document v2 versions

For completeness and as a template for future entries

lukasbestle committed:
Security: New `Urls` field

lukasbestle committed:
Security: Reword coordinated disclosure section

lukasbestle committed:
Security Incidents: Reorder fields for overview

lukasbestle committed:
Security: New `Messages` field

Allows custom security messages in the update check outside of vulnerability fixes

lukasbestle committed:
Security: Cache JSON in CDN for 30 minutes

Ensures quicker responses and takes strain from the getkirby.com server

lukasbestle committed:
Security Versions: Support link field for entries

As a custom replacement for https://getkirby.com/security/end-of-life if needed

distantnative commented:
@lukasbestle let me know when you're ready and need a review

lukasbestle committed:
Security: Rename entry `link` field `status-link`

To make it clearer what the link does

lukasbestle commented:
@distantnative Now it's ready. Probably easiest to review each commit separately for cleaner diffs. Maybe review the [docs commit](https://github.com/getkirby/getkirby.com/pull/1798/commits/de123ae89ad497e8c13c0d46f95877baa7a849ae) first as it makes some of the other changes easier to understand.
